src/components/visualization/EmbeddedViewer.tsx

```diff
@@ -0,0 +1,22 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+export default function PDFViewer({ source }: { source: string }) {
```

Comment:
The function is still named PDFViewer. Can we rename it to something more generic?

Reply:
Good point. Renamed to `EmbeddedViewer()`.